x/mongo/driver/topology/connection_options.go

```diff
@@ -78,6 +80,13 @@ func newConnectionConfig(opts ...ConnectionOption) (*connectionConfig, error) {
 // ConnectionOption is used to configure a connection.
 type ConnectionOption func(*connectionConfig) error
 
+func withTLSConnectionSource(fn func(tlsConnectionSource) tlsConnectionSource) ConnectionOption {
+	return func(c *connectionConfig) error {
+		c.tlsConnectionSource = fn(c.tlsConnectionSource)
+		return nil
+	}
+}
+
```

Comment:
To check I understand, this is a package private function just for the purpose of mocking the `tlsConnectionSource`, right?

Reply:
Yeah, this is an option. Our topology/server/connection configuration uses functional options, so you declare a new connection like:

```
newConnection(
    WithConnectTimeout(10*time.Second),
    WithCompressors([]string{"snappy"}),
    ...
)
```

where each `With*` function is an option. This option is private because we don't actually want people to use it. If it's not specified, `newConnectionConfig` will use `defaultTLSConnectionSource` and our tests can set this option to mock if needed.